opacus/privacy_engine.py

```diff
@@ -155,6 +159,10 @@ def __init__(
 
         self.module = GradSampleModule(module)
 
+        # TODO: add the hooks later? Check which parameters are accessed
+        if isinstance(module, torch.nn.parallel.DistributedDataParallel):
+            self.module.add_ddp_hook(engine=self)
```

Comment:
Only do this if clip_per_layer=True I guess?

Reply:
Yes, indeed! I fixed this